src/reducers/newQuoteReducer.js

```diff
@@ -0,0 +1,19 @@
+import {NEW_QUOTE} from '../actions';
+
+// initial state
+const initState ={
+    quote:'',
+    author:''
+}
+// create reducer
+const newQuoteReducer=(preState=initState,action)=>{
+    switch(action.type){
+        case NEW_QUOTE:
+            console.log(action)
+            const {quote ,author} = action.payload
+            return { quote,author }
```

Comment:
**We don't mutate the state.** You can create a copy with Object.assign(). `Object.assign( {} ,state, { quote, author })` **Note:** You must supply an empty object as the first parameter or it will mutate the first argument. You can also enable the object spread operator proposal to write `{ ...state, quote, author }` instead. More information on: [https://redux.js.org/basics/reducers](url)